### PR TITLE
Define schemas NRS plankton products

### PR DESCRIPTION
**Pull request by mhidas**
This PR adds basic schema definitions for all the raw inputs to be harvested from CSIRO to allow the NRS plankton products to be generate at AODN.

### CODE REVIEW

**Comment by mhidas** [comment]
@clairedavies @mmiller70 @sedgar FYI

**Comment by mhidas** [comment]
Note that there is some automatic validation triggered every time a change is committed to this repository: https://github.com/aodn/public-schema/pull/1/checks?check_run_id=2489540911
This is just checking that the _schema definitions_ themselves (i.e. the yaml files) are valid.

It would also be possible to automatically validate the raw data tables (accessed directly from the [PlanktonTeam's repository](https://github.com/PlanktonTeam/IMOS_Toolbox/tree/master/Plankton/RawData), or eventually from the CSIRO Geoserver). However, I'm not sure this make sense. This should be done the other way around, i.e. agree on the schema definitions, then validate the raw data against them.

File: imos_bgc_db/NRS_Phyto_Raw.schema.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+fields:
+  - name: SAMPLE
+    description: Sample id (references NRS_Samp table)
+    type: integer
```
Comment (mhidas) [inline imos_bgc_db/NRS_Phyto_Raw.schema.yaml:4]: The minimum we need to specify for each field is `name` and `type` (though type defaults to string if not specified).
`description` is just for internal documentation. It is not used for validation, so not essential.

File: imos_bgc_db/NRS_Phyto_Raw.schema.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  - name: CELL_L
+    description: Number of cells per Litre
+    type: number
+  - name: BIOVOLUME_UM3L
+    description: Biovolume (cubic microns per Litre)
+    type: number
+primaryKey:
+  - SAMPLE
+  - TAXON_NAME
```
Comment (mhidas) [inline imos_bgc_db/NRS_Phyto_Raw.schema.yaml:23]: These fields are not allowed to have null values, and each combination needs to be unique.

File: imos_bgc_db/NRS_Phyto_Raw.schema.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+missingValues:
+  - ''
+  - (null)
```
Comment (mhidas) [inline imos_bgc_db/NRS_Phyto_Raw.schema.yaml:26]: We can remove this if we decide to just represent nulls with empty strings (which is the default for Table Schema)

File: imos_bgc_db/NRS_StationInfo.resource.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
+encoding: utf-8
```
Comment (mhidas) [inline imos_bgc_db/NRS_StationInfo.resource.yaml:1]: I've only included this file as an example to show that we can specify more than just the schema, which _may_ be useful at some point (see https://specs.frictionlessdata.io/tabular-data-resource). Note that this now refers to a specific file at the location specified by the `path`.

File: imos_bgc_db/NRS_StationInfo.schema.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    description: IMOS Node operating the station
+    type: string
+  - name: STATECODE
+    description: Australian state the station is located in
+    type: string
+  - name: LONGITUDE
+    description: Longitude (decimal degrees East)
+    type: number
+    maximum: 180
+    minimum: -180
```
Comment (mhidas) [inline imos_bgc_db/NRS_StationInfo.schema.yaml:20]: We can also specify [constraints](https://specs.frictionlessdata.io/table-schema/#constraints) like this, but unfortunately `frictionless` doesn't seem to automatically check these when validating.

**Comment by sedgar** [comment]
Hi Marty, thanks for woking through these. Given it's early days, I'll approve, and we can work through any changes as separate pull requests.

**Review by sedgar — COMMENTED** [review]
We'll work through any necessary changes as separate requests.

**Review by sedgar — COMMENTED** [review]
We'll work through any necessary changes as separate requests.

**Comment by sedgar** [comment]
We'll work through any necessary changes as separate requests.

**Comment by mhidas** [comment]
Thanks @sedgar , that's fine. I don't mind if we don't merge in the schema files I've created, they can just stay here as an example. 
But I'm a bit confused - did you actually mean to approve & merge this PR?

**Comment by sedgar** [comment]
Thanks @mhidas, I just approved back to master branch.

**Comment by mhidas** [comment]
@sedgar I'm going to merge this one because there's a few changes in here I think we should keep. The NRS_*.yaml schema definitions seem to be obsolete now, but they're just an example of what we need. We can remove/update them in the next PR.

**Pull request merged**